src/plugins/vis_types/vega/public/data_model/search_api.ts

```diff
@@ -95,6 +95,17 @@ export class SearchAPI {
                 }
               )
               .pipe(
+                catchError((err) => {
+                  // catch the error and log it into the inspector
+                  this.inspectSearchResult(
+                    {
+                      rawResponse: 'err' in err ? err.err : undefined,
+                    },
+                    requestResponders[requestId]
+                  );
+                  // now throw it again as the Vega parser will catch it and carry on with its flow
+                  return throwError(err);
```

Comment:
nit: what if we continue to inspect search results as a side effects without modifying the main stream. I don't insist but I think the following code will be more semantically right 

 ```
 tap(
                  (data) => this.inspectSearchResult(data, requestResponders[requestId]),
                  (err) =>
                    this.inspectSearchResult(
                      {
                        rawResponse: 'err' in err ? err.err : undefined,
                      },
                      requestResponders[requestId]
                    )
                )
```